cli/fossilize_replay.cpp

```diff
@@ -2554,38 +2546,54 @@ struct ThreadedReplayer : StateCreatorInterface
 		// it's not necessarily complete yet!
 		bool is_outside_range = pipelines.count(h) == 0;
 
-		if (is_outside_range)
+		if (is_outside_range && parsed_parents.count(h) == 0)
 		{
+			{
+				lock_guard<mutex> holder(hash_lock);
+				if (parents.count(h) != 0) {
+					auto &parent = parents.at(h);
+					parent.depth = std::max(parent.depth, depth);
```

Comment:
I still don't think this works, because if parent has dependencies, that chain of dependencies must also be moved up in depth accordingly.

I'm not sure we can avoid doing the traversal after all pipelines have been parsed.

Reply:
Also, brace on next line.

Reply:
For example, consider A (depth 1) -> B (depth 2) -> C (depth 3).
If D -> E -> F -> B, B needs to be adjusted to depth 4, but it also means that C must be updated to depth 5. Doing it recursively inline is a potential exponential explosion in complexity, so that's not really an option I think.

Reply:
Right, not sure how I expected this to work in the first place. I have a local fixup that parses all parents, gathers all top level ones and recursively computes their patents' depth.